provider/lpseal/task_movestorage.go

```diff
@@ -0,0 +1,155 @@
+package lpseal
+
+import (
+	"context"
+
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/lib/harmony/harmonydb"
+	"github.com/filecoin-project/lotus/lib/harmony/harmonytask"
+	"github.com/filecoin-project/lotus/lib/harmony/resources"
+	"github.com/filecoin-project/lotus/provider/lpffi"
+	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+)
+
+type MoveStorageTask struct {
+	sp *SealPoller
+	sc *lpffi.SealCalls
+	db *harmonydb.DB
+
+	max int
+}
+
+func NewMoveStorageTask(sp *SealPoller, sc *lpffi.SealCalls, db *harmonydb.DB, max int) *MoveStorageTask {
+	return &MoveStorageTask{
+		max: max,
+		sp:  sp,
+		sc:  sc,
+		db:  db,
+	}
+}
+
+func (m *MoveStorageTask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (done bool, err error) {
+	var tasks []struct {
+		SpID         int64 `db:"sp_id"`
+		SectorNumber int64 `db:"sector_number"`
+		RegSealProof int64 `db:"reg_seal_proof"`
+	}
+
+	ctx := context.Background()
+
+	err = m.db.Select(ctx, &tasks, `
+		select sp_id, sector_number, reg_seal_proof from sectors_sdr_pipeline where task_id_move_storage=$1`, taskID)
+	if err != nil {
+		return false, xerrors.Errorf("getting task: %w", err)
+	}
+	if len(tasks) != 1 {
+		return false, xerrors.Errorf("expected one task")
+	}
+	task := tasks[0]
+
+	sector := storiface.SectorRef{
+		ID: abi.SectorID{
+			Miner:  abi.ActorID(task.SpID),
+			Number: abi.SectorNumber(task.SectorNumber),
+		},
+		ProofType: abi.RegisteredSealProof(task.RegSealProof),
+	}
+
+	err = m.sc.MoveStorage(ctx, sector)
+	if err != nil {
+		return false, xerrors.Errorf("moving storage: %w", err)
+	}
+
+	_, err = m.db.Exec(ctx, `update sectors_sdr_pipeline set after_move_storage=true where task_id_move_storage=$1`, taskID)
+	if err != nil {
+		return false, xerrors.Errorf("updating task: %w", err)
+	}
+
+	return true, nil
+}
+
+func (m *MoveStorageTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+
+	ctx := context.Background()
+	/*
```

Comment:
Decided to keep it simple for now, but we can definitely make it smarter:
* This is arguably good enough for 90% of setups today as most don't have long-term storage in sealing machines
  * You can mount networked long-term storage, but most SPs have pretty bad experience with doing things that way, mostly because almost all ways to mount remote storage are actually pretty bad at basic things when working under high load, like moving terabytes of data constantly
    * NFS is especially bad, if you move 32G blob of data the whole mount will be unusable until the move finishes; even statfs will hang..
* Reasonable smarts would include:
  * As in the todo below, yield if we're already doing moves
  * Yield based on some storage balancing strategy (maybe even have some additional mechanism governing which storage paths "want" sectors)
  * yield based on "distance" to data source

Imo all of those are just nice to have, and not having those things won't make this worse that lotus-miner - and I'd really rather land this PR and put anything nice-to-have into an issue so we can tackle this at a later stage when other critical things work already